### PR TITLE
Refactor method lookup methods

### PR DESCRIPTION
**Pull request by hgschmie**
There is a random helper in Handlers, move that to a core class and use it across the codebase.

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/core/internal/JdbiClassUtils.java [hunk 1]
```diff
@@ -22,8 +28,32 @@ public static boolean isPresent(String klass) {
         try {
             Class.forName(klass);
             return true;
-        } catch (ReflectiveOperationException ignored) {
+        } catch (ReflectiveOperationException | SecurityException ignored) {
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/core/internal/JdbiClassUtils.java:31]: How can you get a `SecurityException` here? From my reading of the `Class.forName` javadoc, this does not seem possible.

File: core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java [hunk 2]
```diff
@@ -37,11 +37,11 @@ public class OnDemandExtensions implements JdbiConfig<OnDemandExtensions> {
...
-            EQUALS_METHOD = Object.class.getMethod("equals", Object.class);
-            HASHCODE_METHOD = Object.class.getMethod("hashCode");
-            TOSTRING_METHOD = Object.class.getMethod("toString");
-        } catch (NoSuchMethodException wat) {
-            throw new IllegalStateException("OnDemandExtensions initialization failed", wat);
+            EQUALS_METHOD = JdbiClassUtils.methodLookup(Object.class, "equals", Object.class);
+            HASHCODE_METHOD = JdbiClassUtils.methodLookup(Object.class, "hashCode");
+            TOSTRING_METHOD = JdbiClassUtils.methodLookup(Object.class, "toString");
+        } catch (Exception e) {
+            throw new ExceptionInInitializerError(e);
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java:44]: Is this catch necessary at all now? `methodLookup` declares no checked exceptions

File: core/src/test/java/org/jdbi/v3/core/extension/TestExtensionContext.java [hunk 3]
```diff
@@ -59,8 +60,8 @@ public interface TestExtension {
         Handle getHandle();
 
         default void checkContextInDefaultMethod() throws Exception {
-            Method m = TestExtension.class.getMethod("checkContextInDefaultMethod");
-            // there is not extension method set in default methods. These methods
+            Method m = JdbiClassUtils.methodLookup(TestExtension.class, "checkContextInDefaultMethod");
```
Comment (stevenschlansker) [inline core/src/test/java/org/jdbi/v3/core/extension/TestExtensionContext.java:63]: This change seems to add another layer of abstraction to this code without really making a specific improvement.

**Review by stevenschlansker — APPROVED** [review]

File: core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java (hunk 2 above)
Comment (hgschmie) [reply core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java:44]: catch Exception also catches the runtime exception and so this throws ExceptionInInitializerError (which is the right thing to do in an initializer)

File: core/src/test/java/org/jdbi/v3/core/extension/TestExtensionContext.java (hunk 3 above)
Comment (hgschmie) [reply core/src/test/java/org/jdbi/v3/core/extension/TestExtensionContext.java:63]: mechanic search-and-replace. :-)

File: core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java (hunk 2 above)
Comment (stevenschlansker) [reply core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java:44]: That wrapping is already done by the compiler if you just leave the catch out, pretty sure.

**Pull request merged**